modules/Microsoft.Sql/servers/databases/deploy.bicep

```diff
@@ -158,6 +158,27 @@ param isLedgerOn bool = false
 @description('Optional. Maintenance configuration ID assigned to the database. This configuration defines the period when the maintenance updates will occur.')
 param maintenanceConfigurationId string = ''
 
+@description('Optional. Differential backup interval in hours.')
+param backupDifferentialInterval int = 24
+
+@description('Optional. Poin-in-time backup retention in days.')
+@maxValue(35)
+param backupLogRetentionDays int = 7
+
+@description('Optional. Monthly retention in ISO 8601 duration format.')
+param backupWeeklyRetention string = ''
+
+@description('Optional. Weekly retention in ISO 8601 duration format.')
+param backupMonthlyRetention string = ''
+
+@description('Optional. Backup week of year to keep for yearly retention.')
+@minValue(1)
+@maxValue(53)
+param backupYearlyRetentionWeek int = 1
+
+@description('Optional. Yearly retention in ISO 8601 duration format.')
+param backupYearlyRetention string = ''
+
```

Comment:
What if we also create objects of this and use `contains(objectName, propertyName) ? use property : use default value` in the module declaration itself? 

So the object parameters would be `shortTermBackupRetentionPolicy` and `longTermBackupRetentionPolicy` and the properties as 1:1 match with Azure Resource Reference.

Reply:
Good idea ([ref](https://github.com/Azure/ResourceModules/blob/main/modules/Microsoft.Storage/storageAccounts/deploy.bicep#:~:text=//%20Containers-,module%20storageAccount_blobServices%20%27blobServices/deploy.bicep%27%20%3D%20if%20(!empty(,%7D,-//%20File%20Shares))

Reply:
Re-structured into an object per sub-deployment, please have a look.